### PR TITLE
community: update AzureSearch class to work with azure-search-documents=11.4.0

### PR DESCRIPTION
**Pull request by lz-chen**
- **Description:** Updates `libs/community/langchain_community/vectorstores/azuresearch.py` to support the stable version `azure-search-documents=11.4.0` 
  - **Issue:** https://github.com/langchain-ai/langchain/issues/14534, https://github.com/langchain-ai/langchain/issues/15039, https://github.com/langchain-ai/langchain/issues/15355
  - **Dependencies:** azure-search-documents>=11.4.0

### CODE REVIEW

**Comment by lz-chen** [comment]
Hello, there is a previous PR [#13472]( https://github.com/langchain-ai/langchain/pull/13472) that provided some partial fix to the `AzureSearch` class but did not fully update the dependency. Now calling class as vector store still doesn't work in my case, with the problems encontered  by others in the issues mentioned in the description. 
I'd like to contribute to update the dependency to `azure-search-documents=11.4.0`. However I am not sure if that means updating all the `pyproject.toml` and `poetry.lock` files that has `azure-search-documents`? What is the best way to do that? @efriis @hwchase17

**Comment by efriis** [comment]
Hey @lz-chen ! You actually can't do this with a community integration, as it's not a concrete dependency. Unfortunately, people need to manage their other python package versions themselves. 

A good addition would be a note in the documentation to upgrade that package!

**Comment by lz-chen** [comment]
@efriis Thanks for the reply 😊 What does that mean for this PR? Should I keep the changes I made in `libs/community/langchain_community/vectorstores/azuresearch.py` without updating the dependencies in `libs/langchain/pyproject.toml` and `libs/langchain/poetry.lock`? 
Or follow what's done in PR [#13472]( https://github.com/langchain-ai/langchain/pull/13472) to keep support for both `azure-search-documents>=11.4.0` and `azure-search-documents<11.4.0`?

**Comment by HeidiSteen** [comment]
@mattgotteiner for subject matter expertise on Azure Search SDK migration to 11.4 stable library.  Agree with @lz-chen that the existing vector store integration module doesn't work with 11.4.0.

**Comment by Skar0** [comment]
Hi @lz-chen,

Thanks for merging my PR into your update! I reviewed the original changes and tested them with `azure-search-documents==11.4.0`, and everything seems to work just fine :) My PR aimed to address a specific bug related to the removal of `query_language` in `azure-search-documents==11.4.0`. I'm happy to discuss any further details or assist with additional improvements if needed.

**Comment by konradbjk** [comment]
Looking forward to getting this merged. Due to this issue I cannot use Azure Search as a vector store

**Comment by mattgotteiner** [comment]
Thank you for making this update - the changes look good

File: libs/community/langchain_community/vectorstores/azuresearch.py [hunk 1]
```diff
@@ -155,77 +140,56 @@ def fmt_err(x: str) -> str:
...
-                                ],
-                            ),
-                        )
-                    ]
-                )
+        if semantic_configurations is None and semantic_configuration_name is not None:
+            semantic_configuration = SemanticConfiguration(
+                name=semantic_configuration_name,
+                prioritized_fields=SemanticPrioritizedFields(
+                    # title_field=SemanticField(field_name="title"),
```
Comment (pamelafox) [inline libs/community/langchain_community/vectorstores/azuresearch.py:180]: Why are these 2 lines commented out? Should they be deleted?

**Review by pamelafox — APPROVED** [review]
Looks good! I assume there are no tests that need updating. (GitHub shows me as a reviewer but an actual maintainer will need to review and approve workflows)

**Comment by pamelafox** [comment]
Just looking at the discussion here about pyproject.toml-

It looks like the main langchain pyproject.toml doesn't reference azure-search-documents, which makes sense as this in community. However, I notice that the langchain-community pyproject.toml also doesn't reference azure-search-documents. I believe that it probably should, just based off similar recent integration updates, like this PR for Azure Document Intelligence:
https://github.com/langchain-ai/langchain/pull/14389

The file to change:
https://github.com/langchain-ai/langchain/blob/ac970c9497e2aca1f6396c3f6954b4f6cd0ac879/libs/community/pyproject.toml#L86
Notably it'd be marked as optional=True, like the other community dependencies.

That's my suggestion after looking through how the packages are structured. Perhaps @efriis could confirm. Thanks!

**Comment by baskaryan** [comment]
> Just looking at the discussion here about pyproject.toml-
> 
> It looks like the main langchain pyproject.toml doesn't reference azure-search-documents, which makes sense as this in community. However, I notice that the langchain-community pyproject.toml also doesn't reference azure-search-documents. I believe that it probably should, just based off similar recent integration updates, like this PR for Azure Document Intelligence: #14389
> 
> The file to change:
> 
> https://github.com/langchain-ai/langchain/blob/ac970c9497e2aca1f6396c3f6954b4f6cd0ac879/libs/community/pyproject.toml#L86
> 
> 
> Notably it'd be marked as optional=True, like the other community dependencies.
> That's my suggestion after looking through how the packages are structured. Perhaps @efriis could confirm. Thanks!

Optional dependencies only need to be included in pyproject if they're required for unit tests, so no need to add in this case!

**Review by baskaryan — APPROVED** [review]

**Comment by pamelafox** [comment]
@baskaryan Ah, thanks for clarifying, that explains why azure-document-intelligence was added in that PR, as it had a short unit test.

**Comment by mattgotteiner** [comment]
We are also missing an `from azure.search.documents.indexes.models import VectorSearch`

**Comment by mattgotteiner** [comment]
the import is only performed with `typing.TYPE_CHECKING` is true

**Comment by kristapratico** [comment]
@mattgotteiner noticed these issues today as well and opened a PR to fix here: https://github.com/langchain-ai/langchain/pull/17599

File: libs/community/langchain_community/vectorstores/azuresearch.py [hunk 2]
```diff
@@ -451,39 +413,30 @@ def vector_search_with_score(
...
         docs = [
             (
                 Document(
                     page_content=result.pop(FIELDS_CONTENT),
-                    metadata={
-                        **(
-                            {FIELDS_ID: result.pop(FIELDS_ID)}
-                            if FIELDS_ID in result
-                            else {}
-                        ),
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/azuresearch.py:477]: hey @lz-chen @mattgotteiner, was there a specific reason we removed FIELDS_ID in this pr?

**Pull request merged**